app/controllers/api/migrations.php

```diff
@@ -0,0 +1,824 @@
+    ->label('scope', 'migrations.write')
+    ->label('event', 'migrations.create')
+    ->label('audits.event', 'migration.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'migrateNHost')
+    ->label('sdk.description', '/docs/references/migrations/migration-nhost.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_ACCEPTED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION)
+    ->param('resources', [], new ArrayList(new WhiteList(NHost::getSupportedResources())), 'List of resources to migrate')
+    ->param('subdomain', '', new URL(), "Source's Subdomain")
+    ->param('region', '', new Text(512), "Source's Region")
+    ->param('adminSecret', '', new Text(512), "Source's Admin Secret")
+    ->param('database', '', new Text(512), "Source's Database Name")
+    ->param('username', '', new Text(512), "Source's Database Username")
+    ->param('password', '', new Text(512), "Source's Database Password")
+    ->param('port', 5432, new Integer(true), "Source's Database Port", true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (array $resources, string $subdomain, string $region, string $adminSecret, string $database, string $username, string $password, int $port, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
+        $migration = $dbForProject->createDocument('migrations', new Document([
+            '$id' => ID::unique(),
+            'status' => 'pending',
+            'stage' => 'init',
+            'source' => NHost::getName(),
+            'credentials' => [
+                'subdomain' => $subdomain,
+                'region' => $region,
+                'adminSecret' => $adminSecret,
+                'database' => $database,
+                'username' => $username,
+                'password' => $password,
+                'port' => $port,
+            ],
+            'resources' => $resources,
+            'statusCounters' => '{}',
+            'resourceData' => "{}",
+            'errors' => []
+        ]));
+
+        $eventsInstance->setParam('migrationId', $migration->getId());
+
+        // Trigger Transfer
+        $event = new Migration();
+        $event
+            ->setMigration($migration)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_ACCEPTED)
+            ->dynamic($migration, Response::MODEL_MIGRATION);
+    });
+
+App::get('/v1/migrations/nhost/report')
+    ->groups(['api', 'migrations'])
+    ->desc('Generate a report on NHost Data')
+    ->label('scope', 'migrations.write')
+    // ->label('event', 'migrations.report')
+    // ->label('audits.event', 'migration.report')
```

Comment:
Let's remove this if they're unused, the same applies to other source's endpoints